components/automate-cli/cmd/chef-automate/generate-a2harb.go

```diff
@@ -69,3 +81,186 @@ func renderSettingsToA2HARBFile(templateName string, data interface{}) string {
+		return errors.New("Invalid or empty aws region")
+	} else if len(awsConfig.Aws.Config.SSHKeyPairName) < 1 {
+		return errors.New("Invalid or empty aws ssh_key_pair_name")
+	} else if len(awsConfig.Aws.Config.AutomateServerInstanceType) < 1 {
+		return errors.New("Invalid or empty aws automate_server_instance_type")
+	} else if len(awsConfig.Aws.Config.ChefServerInstanceType) < 1 {
+		return errors.New("Invalid or empty aws chef_server_instance_type")
+	} else if len(awsConfig.Aws.Config.ElasticsearchServerInstanceType) < 1 {
+		return errors.New("Invalid or empty aws elasticsearch_server_instance_type")
+	} else if len(awsConfig.Aws.Config.PostgresqlServerInstanceType) < 1 {
+		return errors.New("Invalid or empty aws postgresql_server_instance_type")
+	} else if len(awsConfig.Aws.Config.AutomateLbCertificateArn) < 1 {
+		return errors.New("Invalid or empty aws automate_lb_certificate_arn")
+	} else if len(awsConfig.Aws.Config.ChefServerLbCertificateArn) < 1 {
+		return errors.New("Invalid or empty aws chef_server_lb_certificate_arn")
+	} else if len(awsConfig.Aws.Config.AutomateEbsVolumeIops) < 1 {
+		return errors.New("Invalid or empty aws automate_ebs_volume_iops")
+	} else if len(awsConfig.Aws.Config.AutomateEbsVolumeSize) < 1 {
+		return errors.New("Invalid or empty aws automate_ebs_volume_size")
+	} else if len(awsConfig.Aws.Config.AutomateEbsVolumeType) < 1 {
+		return errors.New("Invalid or empty aws automate_ebs_volume_type")
+	} else if len(awsConfig.Aws.Config.ChefEbsVolumeIops) < 1 {
+		return errors.New("Invalid or empty aws chef_ebs_volume_iops")
+	} else if len(awsConfig.Aws.Config.ChefEbsVolumeSize) < 1 {
+		return errors.New("Invalid or empty aws chef_ebs_volume_size")
+	} else if len(awsConfig.Aws.Config.ChefEbsVolumeType) < 1 {
+		return errors.New("Invalid or empty aws chef_ebs_volume_type")
+	} else if len(awsConfig.Aws.Config.ElasticsearchEbsVolumeIops) < 1 {
+		return errors.New("Invalid or empty aws elasticsearch_ebs_volume_iops")
+	} else if len(awsConfig.Aws.Config.ElasticsearchEbsVolumeSize) < 1 {
+		return errors.New("Invalid or empty aws elasticsearch_ebs_volume_size")
+	} else if len(awsConfig.Aws.Config.ElasticsearchEbsVolumeType) < 1 {
+		return errors.New("Invalid or empty aws elasticsearch_ebs_volume_type")
+	} else if len(awsConfig.Aws.Config.PostgresqlEbsVolumeIops) < 1 {
+		return errors.New("Invalid or empty aws postgresql_ebs_volume_iops")
+	} else if len(awsConfig.Aws.Config.PostgresqlEbsVolumeSize) < 1 {
+		return errors.New("Invalid or empty aws postgresql_ebs_volume_size")
+	} else if len(awsConfig.Aws.Config.PostgresqlEbsVolumeType) < 1 {
+		return errors.New("Invalid or empty aws postgresql_ebs_volume_type")
+	} else {
+		return nil
+	}
+}
+
+func validateIPs(exitingInfra ExistingInfraConfigToml) error {
+	for _, element := range exitingInfra.ExistingInfra.Config.AutomateIps {
+		if checkIPAddress(element) != nil {
+			return errors.New("Automate Ip " + element + "is not valid")
```

Comment:
in case of public ip missing ....add err msg .....use private ip ...in public ip place